library/src/main/java/com/datatorrent/lib/io/jms/JMSBase.java

```diff
@@ -129,12 +173,20 @@ public void setConnectionFactoryClass(String connectionFactoryClass)
    */
   public Map<String, String> getConnectionFactoryProperties()
   {
-    return connectionFactoryProperties;
+    if (connectionFactoryBuilder instanceof DefaultConnectionFactoryBuilder) {
```

Comment:
On further thought about the serialization aspects and the extra objects, it would create, suggest having the connectionFactoryBuilder null by default and creating it in the helper method. So if connectionFactoryBuilder is null or not instance of DefaultConnectionFactoryBuilder set it to a new DefaultConnectionFactoryBuilder. If someone sets a different factory builder then an extra default builder will not be used.